js/ag_reference.js

```diff
@@ -0,0 +1,299 @@
+/*
+ * - Column and Refernce have a lot of arguments, I don't know how to fix that.
+ *   The same way we have the location object, we can have something like that here,
+ *   which has the service, catalog, path, sort, paging, and search in it. but should we?
+ *
+ * - Search is currently based on just one column. It's now a object of `column` and `term`,
+ *   to make it easier to pass and also be able to pass default search value.
+ *
+ * - chaise recordset is using the reference.location.searchTerm to get the search term. That's why
+ *   I added the location object.
+ *
+ * - read is going to create the uri, it should use the location, keyColumns and aggregateColumns to do so.
+ *   It's better if location could have functions to create the uri, like Location prototype in Reference.
+ *   but how?
+ *   
+ */
+
+function AttributeGroupReference(keyColumns, aggregateColumns, location) {
+    
+    /**
+     * Array of AttributeGroupColumn that will be used as the key columns
+     * @type {ERMrest.AttributeGroupColumn[]}
+     */
+    this._keyColumns = keyColumns;
+    
+    /**
+     * Array of AttributeGroupColumn that will be used for the aggregate results
+     * @type {ERMrest.AttributeGroupColumn[]}
+     */
+    this._aggregateColumns = aggregateColumns;
+    
+    this._location = location;
+}
+AttributeGroupReference.prototype = {
+    
+    constructor: AttributeGroupReference,
+    
+    get location () {
+        return this._location;
+    },
+    
+    /**
+     * the displayname of the reference
+     * TODO not sure if this sis needed
```

Comment:
We will need it for when there is a picker popup in the facet.

Maybe use the `table.name` and if that's not defined (chances are it won't), we should have fallback heuristics. Maybe displayname should be related to what aggregate was called?